internal/log/writer.go

```diff
@@ -49,15 +49,28 @@ func Writer(level Level, keysAndValues ...interface{}) io.WriteCloser {
 // level with the supplied keysAndValues.
 //
 // This method will block until r returns an EOF or causes an error.
-func WriteTo(level Level, r io.Reader, keysAndValues ...interface{}) error {
+func WriteTo(level Level, rd io.Reader, keysAndValues ...interface{}) error {
 	logger := defaultLogger.With(keysAndValues...)
-	scanner := bufio.NewScanner(r)
-	for scanner.Scan() {
-		text := scanner.Text()
+	r := bufio.NewReader(rd)
+	for {
+		line, err := r.ReadBytes('\n')
+		l := len(line)
+		// Trim any trailing \n or \r\n from the buffer.
+		if l > 0 && line[l-1] == '\n' {
```

Comment:
Could we just use https://pkg.go.dev/bytes#TrimRight here? This would also trim any `\r` that isn't followed by a `\n` but that seems OK.

Reply:
Good idea. Swapped for `TrimRightFunc`